src/french.js

```diff
@@ -1,68 +1,434 @@
+          ! 'on' and objects 'can' also have a gender !
+      }
+  */
   conjugate (word, info) {
-    // Format for info: {tense: 'present', formal: 'true/false', gender: 'female/male/undefined', singular: 'true/false/undefined', person: '1'/'2'/'3'}
-    //TODO define 3 verb group types
+    switch (info.mood) {
+      case 'Indicatif':
+        switch (info.tense) {
+          case 'Présent':
+            return this.doIndicatifPresent(word, info);
+            break;
+          case 'Passé Composé':
+            break;
+          case 'Imparfait':
+            break;
+          case 'Plus-que-parfait':
+            break;
+          case 'Passé Simple':
+            break;
+          case 'Passé Antérieur':
+            break;
+          case 'Futur Simple':
+            break;
+          case 'Futur Antérieur':
+            break;
+          default:
+            return 'Indicatif Tense not found';
+        }
+        break;
+      case 'Infinitif':
+        switch (info.tense) {
+          case 'Présent':
+           return word;
+           break;
+          case 'Passé':
+            break;
+          default:
+            return 'Infinitif Tense not found';
+        }
+        break;
+      case 'Participe':
+        switch (info.tense) {
+          case 'Présent':
+           break;
+          case 'Passé':
+            break;
+          default:
+            return 'Participe Tense not found';
+        }
+        break;
+      case 'Subjonctif':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé':
+            break;
+          case 'Imparfait':
+            break;
+          case 'Plus-que-parfait':
+            break;
+          default:
+            return 'Subjonctif Tense not found';
+        }
+        break;
+      case 'Impératif':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé':
+            break;
+          default:
+            return 'Impératif Tense not found';
+        }
+        break;
+      case 'Conditionnel':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé 1':
+            break;
+          case 'Passé 2':
+            break;
+          default:
+          return 'Conditionnel Tense not found';
+        }
+        break;
+      default:
+        return 'Mood not found';
+    }
+  }
 
-   switch (info.tense) {
-     case 'present':
-       return this.doPresent(word, info);
-       break;
-     default:
-       return 'Could not find any rules for conjugation';
-   }
+  findSubject (info) {
+    // returns the subject for conjugation: 's1'/'s2'/'s3'/'p1'/'p2'/'p3'
+    if (info.singular === true) {
+      if (info.person === '2' && info.formal === true){
+        //if the subject ('you') is meant as formal and singular, it is conjugated as plural, but the past participle stays singular
+        return 'p2';
+      }
+      return 's' + info.person;
```

Comment:
I changed this to `s${info.person}`and changed some others too.
There were some cases where I wasn't sure if I should use a template literal or If I should just leave the expression as is.
like : return [stem.slice(0, -3) + 'èvr' + ending];
When do you know if you should use a template literal or avoid it?